src/v/utils/xid.h

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright 2024 Redpanda Data, Inc.
+ *
+ * Use of this software is governed by the Business Source License
+ * included in the file licenses/BSL.md
+ *
+ * As of the Change Date specified in that file, in accordance with
+ * the Business Source License, use of this software will be governed
+ * by the Apache License, Version 2.0
+ */
+
+#pragma once
+
+#include "base/seastarx.h"
+#include "bytes/bytes.h"
+
+#include <seastar/core/sstring.hh>
+
+#include <fmt/core.h>
+
+#include <array>
+#include <cstdint>
+#include <exception>
+
+class invalid_xid final : public std::exception {
+public:
+    explicit invalid_xid(const ss::sstring&);
+    const char* what() const noexcept final { return _msg.c_str(); }
+
+private:
+    ss::sstring _msg;
+};
+
+/**
+ * Class representing XID
+ *
+ * Based on (https://github.com/rs/xid)
+ */
+class xid {
```

Comment:
why not look at `bytes class` for this. 

`xid = typedef ss::basic_string<uint8_t, 12>`